src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -228,31 +231,46 @@
    {:on-click #(dispatch! "/#/administration/catalogue-items")}
    (text :t.administration/cancel)])
 
-(defn- save-catalogue-item-button []
+(defn- save-catalogue-item-button [on-click]
   (let [form @(rf/subscribe [::form])
         request (build-request form)]
     [:button.btn.btn-primary
-     {:on-click #(rf/dispatch [::create-catalogue-item request])
+     {:on-click #(on-click request)
       :disabled (nil? request)}
      (text :t.administration/save)]))
 
 (defn create-catalogue-item-page []
-  (let [loading? (rf/subscribe [::loading?])]
-    [:div
-     [administration-navigator-container]
-     [:h2 (text :t.administration/create-catalogue-item)]
-     [collapsible/component
-      {:id "create-catalogue-item"
-       :title (text :t.administration/create-catalogue-item)
-       :always [:div
-                (if @loading?
-                  [:div#catalogue-item-loader [spinner/big]]
-                  [:div#catalogue-item-editor
-                   [catalogue-item-title-field]
-                   [catalogue-item-workflow-field]
-                   [catalogue-item-resource-field]
-                   [catalogue-item-form-field]
-
-                   [:div.col.commands
-                    [cancel-button]
-                    [save-catalogue-item-button]]])]}]]))
+  (let [loading? (rf/subscribe [::loading?])
+        state (r/atom nil)
+        on-pending #(reset! state {:status :pending})
+        on-success #(reset! state {:status :saved})
+        on-error #(reset! state {:status :failed :error %})
+        on-modal-close #(do (reset! state nil)
+                            (dispatch! "#/administration/catalogue-items"))]
```

Comment:
If the submit fails, this will move the user away from this page. On error the user should stay on the same page so that he can fix any errors and try again.